realm/realm-library/src/syncIntegrationTest/java/io/realm/SyncedRealmIntegrationTests.java

```diff
@@ -299,7 +293,7 @@ public void waitForInitialRemoteData_readOnlyTrue_throwsIfWrongServerSchema() {
             // schema.
             realm = Realm.getInstance(configNew);
             fail();
-        } catch (IllegalStateException ignored) {
+        } catch (RealmMigrationNeededException ignored) {
```

Comment:
Hmm. this is confusing `RealmMigrationNeededException` doesn't make sense in an additive Sync schema scenario...

Reply:
Good point. I'll fix it.

Reply:
There is something odd here. But changing the exception to `IllegalState` causes other errors. It looks like a difference between query-based and fully synchronized Realms (since this PR changes the default back to fully synchronized). I have created https://github.com/realm/realm-java/issues/6759 to track it.